celery/backends/elasticsearch.py

```diff
@@ -225,21 +249,33 @@ def mget(self, keys):
         return [self.get(key) for key in keys]
 
     def delete(self, key):
-        self.server.delete(index=self.index, doc_type=self.doc_type, id=key)
+        if self.doc_type:
+            self.server.delete(index=self.index, id=key, doc_type=self.doc_type)
+        else:
+            self.server.delete(index=self.index, id=key)
 
     def _get_server(self):
         """Connect to the Elasticsearch server."""
         http_auth = None
         if self.username and self.password:
             http_auth = (self.username, self.password)
-        return elasticsearch.Elasticsearch(
-            f'{self.host}:{self.port}',
-            retry_on_timeout=self.es_retry_on_timeout,
-            max_retries=self.es_max_retries,
-            timeout=self.es_timeout,
-            scheme=self.scheme,
-            http_auth=http_auth,
-        )
+        if elasticsearch.VERSION[0] <= 7:
```

Comment:
less or equal 7 or 8?

Reply:
I'll do some digging on the client side to see what specific version they removed the scheme parameter from. But I know the 8.9 client version I installed did not have it where the 7.17 client did. So I assumed that 7.x and below would require the scheme parameter. I'll also check if I supply it as part of the URI with the 7.x client if I can skip the scheme entirely and not have that if check at all.

Reply:
we can try keep support of both 7.x and 8.x for better upgrade ability